src/catchup/ApplyLedgerWork.cpp

```diff
@@ -23,6 +23,16 @@ BasicWork::State
 ApplyLedgerWork::onRun()
 {
     ZoneScoped;
+    if (mApp.getLedgerManager().isBooting())
+    {
+        mApp.postOnBackgroundThread(
```

Comment:
By the time we get to `applyLedger`, we should never be in rebuilding state. Applying ledger enforces that the state is complete and valid. Let's implement the wait in specific places, where we anticipate core to be in rebuilding state. Specifically, on startup (loadLastKnownLedger) and when catchup is done (setLastClosedLedger).